Objects/longobject.c

```diff
@@ -42,7 +43,7 @@ static inline void
 _Py_DECREF_INT(PyLongObject *op)
 {
     assert(PyLong_CheckExact(op));
-    _Py_DECREF_SPECIALIZED((PyObject *)op, (destructor)PyObject_Free);
+    _Py_DECREF_SPECIALIZED((PyObject *)op, (destructor) _PyLong_ExactDealloc);
```

Comment:
```suggestion
    _Py_DECREF_SPECIALIZED((PyObject *)op, (destructor)_PyLong_ExactDealloc);
```

I'm also not sure whether you need the cast. If you need the cast, I'd suggest changing the signature of the drstructor itself.